pkg/daemon/update.go

```diff
@@ -823,21 +823,21 @@ func reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) (*machineConfigDif
 			return nil, fmt.Errorf("ignition Passwd Groups section contains changes")
 		}
 		if !reflect.DeepEqual(oldIgn.Passwd.Users, newIgn.Passwd.Users) {
-			if len(oldIgn.Passwd.Users) > 0 && len(newIgn.Passwd.Users) == 0 {
-				return nil, fmt.Errorf("ignition passwd user section contains unsupported changes: user core may not be deleted")
-			}
+			newEmpty := len(newIgn.Passwd.Users) == 0
 			// there is an update to Users, we must verify that it is ONLY making an acceptable
 			// change to the SSHAuthorizedKeys for the user "core"
 			for _, user := range newIgn.Passwd.Users {
 				if user.Name != constants.CoreUserName {
 					return nil, fmt.Errorf("ignition passwd user section contains unsupported changes: non-core user")
 				}
 			}
-
-			glog.Infof("user data to be verified before ssh update: %v", newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1])
-			if err := verifyUserFields(newIgn.Passwd.Users[len(newIgn.Passwd.Users)-1]); err != nil {
-				return nil, err
+			if !newEmpty {
```

Comment:
Nit:  doesn't really matter, but we probably don't need the newEmpty variable, if we were getting value from that variable name we could probably just comment the condition, e.g.:  
```suggestion
                        // We don't want to panic if the "new" users is empty, and it's still reconcilable because the absence of a user here does not mean "remove the user from the system"
			if len(newIgn.Passwd.Users) != 0 {
```
 (feel free to ignore this or use your words rather than my words)